core/lib/dal/src/events_web3_dal.rs

```diff
@@ -127,15 +171,25 @@ impl EventsWeb3Dal<'_, '_> {
             arg_index += 1;
         }
 
+        if api_mode == APIMode::Modern {
+            where_sql += &format!(
+                " AND NOT (address = ${} AND topic1 = ${})",
+                arg_index,
+                arg_index + 1
+            );
```

Comment:
Dumb question: Is speedup from having this as a part of the query that large that it justifies breaking encapsulation? IMO, from the encapsulation perspective, it would make sense to leave DAL as-is and perform event filtering and index translation on the server level. Is this infeasible because of the performance degradation?